### ISSUE TITLE
multi-class classification with TFDF: map true labels to probabilities vector of model.predict()

### ISSUE THREAD

**Issue by miroC911** [tensorflow/decision-forests#117]
Hello,
Sorry for this basic question. I am wondering how to map true (original) class labels (strings) from model.predict (test) probability scores. I am retrieving top class index via argmax, and I want to map this index to the original label. Basically I want to get the label string for that top class index...but I am not sure how TFDF models sort the labels to get that info or if I can access them in any way...
I am passing a pandas df to RandomForestModel()  via tfdf.keras.pd_dataframe_to_tf_dataset(df, label).
Thanks
Mireille

**Comment by achoum** [comment]
**Background**

Keras assumes that classification labels are integers in the range [0, num_classes).
To make your life easier, [tfdf.keras.pd_dataframe_to_tf_dataset](https://github.com/tensorflow/decision-forests/blob/main/tensorflow_decision_forests/keras/core.py#L2429) converts string labels into integers. The exact conversion is done [here](https://github.com/tensorflow/decision-forests/blob/main/tensorflow_decision_forests/keras/core.py#L2506).

The label indices are computed after having sorted lexicographically the string label representation.

**Your case**

To make sure you have the correct matching between label string and integer representations, it is best that you do the conversion manually before calling pd_dataframe_to_tf_dataset. This is the option used in the [beginner colab](https://www.tensorflow.org/decision_forests/tutorials/beginner_colab#load_the_dataset_and_convert_it_in_a_tfdataset
).
